tests/integration/integration_suite_test.go

```diff
@@ -191,8 +191,9 @@ func TestMedusaDeploymentScenario(t *testing.T) {
 			namespace := initializeCluster(t)
 			medusaSuccess := t.Run("Test backup and restore", func(t *testing.T) {
 				createMedusaSecretAndInstallDeps(t, namespace, backend)
-				deployClusterForMedusa(t, namespace, backend)
+				deployClusterForMedusa(t, namespace, backend, 1)
 				testMedusa(t, namespace, backend, backupName)
+				scaleUpCassandra(t, namespace, backend, 2)
```

Comment:
Do we need to test the scale up for each backend? The bug that prevented scale up is not impacted by the backend used. On that basis I wonder if it makes sense to have a separate test that focuses on scale up/down with backup/restore.

Reply:
I can make a different test suite with just MinIO, but I'm not sure it's worth the pain of creating yet another test suite. It just takes a few more minutes to complete with both backends (although S3 is mostly not running due to the use of a secret).
As long as we test this behavior, I think it's fine to run it each time we have a backup test.

Reply:
I think John meant that we shouldn't think "backup / restore" as the main thing in that scaling test, but the actual scale up / scale down. The backup restore would just be a minor detail / single subtest in that process. 

The problem with "few minutes more" is that they accumulate and soon it's hours to see if a PR passes at which point no one will give a damn if it passes or not, especially if it even once fails without the PR being the reason.

Reply:
>I think John meant that we shouldn't think "backup / restore" as the main thing in that scaling test, but the actual scale up / scale down. The backup restore would just be a minor detail / single subtest in that process.

I do think we want to implement separate scaling tests and backup/restore should be covered in those tests, but I was also thinking about execution times.

>I can make a different test suite with just MinIO, but I'm not sure it's worth the pain of creating yet another test suite. It just takes a few more minutes to complete with both backends (although S3 is mostly not running due to the use of a secret).
As long as we test this behavior, I think it's fine to run it each time we have a backup test.

I looked at the results for [TestMedusaDeploymentScenario/"Minio"] (https://github.com/k8ssandra/k8ssandra/pull/647/checks?check_run_id=2298520333) in this PR. The test took 721 seconds. 

Then I looked at [TestMedusaDeploymentScenario/"Minio"](https://github.com/k8ssandra/k8ssandra/pull/638/checks?check_run_id=2300598089) for #638. It took 342 seconds.

That is a difference of over 6 minutes. I am sure that there will be a decent bit of variation if we were to consider more samples, but I am confident it is still going to be a very appreciable difference. Because scaling up is slow, that difference is enough for me to disagree that it is fine to run it across each backup/restore test. Also consider this. Right now we are only testing again MinIO and S3, but we will be adding GCS soon and then hopefully Azure after that.

Here is my thinking. We have the additional test coverage which is great. Thanks for that @adejanovski. We want to get a release out asap. The test changes would be non-trivial, and I don't see any reason to rush them. I propose we merge the PR and address test refactoring post 1.1. 

@burmanm and @adejanovski does that sound reasonable?

Reply:
Just a couple of thoughts on things here:

> Here is my thinking. We have the additional test coverage which is great. Thanks for that @adejanovski. We want to get a release out asap. The test changes would be non-trivial, and I don't see any reason to rush them. I propose we merge the PR and address test refactoring post 1.1.

Totally agree with this, the big benefit is the additional testing, and at the moment, that's a really big benefit because it gets us to a higher level of confidence to release 1.1.

Sounds like the refactoring is much about the speed of the tests - wouldn't we, in theory, be capable of running the different backend (s3, minio, etc) tests in parallel - so even if each did include the scale up/down we could maybe reduce the overall impact that way?  Perhaps we're limited there by GH's runners though?

Reply:
> I think John meant that we shouldn't think "backup / restore" as the main thing in that scaling test, but the actual scale up / scale down. The backup restore would just be a minor detail / single subtest in that process.

> I do think we want to implement separate scaling tests and backup/restore should be covered in those tests, but I was also thinking about execution times.

I understand what you're both saying but for this particular case I just wanted to test that scaling up is still possible after a restore.
We'll need separate, proper scaling tests to be implemented, which was not the intent in this PR.

We could restrict the scaling test to just a specific backend, but overall it's going to take the same amount of time as our checks will take as long as the longest of all of them (they're running in parallel, so having a single one running the scale up test won't save time overall). 
We also have other leads to reduce the overall duration, including tuning the probes. I've seen Cassandra pods being restarted multiple times during restores instead of just once, which make the overall operation last way longer.

> Here is my thinking. We have the additional test coverage which is great. Thanks for that @adejanovski. We want to get a release out asap. The test changes would be non-trivial, and I don't see any reason to rush them. I propose we merge the PR and address test refactoring post 1.1.

I agree 👍 

> Sounds like the refactoring is much about the speed of the tests - wouldn't we, in theory, be capable of running the different backend (s3, minio, etc) tests in parallel - so even if each did include the scale up/down we could maybe reduce the overall impact that way? Perhaps we're limited there by GH's runners though?

It's the case already, we're running the tests in parallel, and have enough GH runners so that they all run at the same time.

I agree with Michael and John on trying to avoid making tests last longer, and I hope we can focus on speeding things up in the coming sprints while still allowing to add test cases. We'll need to do some profiling on K8ssandra deployments to see where time is spent and tune appropriately (which John already started doing).

Reply:
>Sounds like the refactoring is much about the speed of the tests - wouldn't we, in theory, be capable of running the different backend (s3, minio, etc) tests in parallel - so even if each did include the scale up/down we could maybe reduce the overall impact that way? Perhaps we're limited there by GH's runners though?

@jdonenine We can run the tests in parallel which does mitigate the impact; however, it means those runners are tied up longer before they can move onto the next tests. There is another aspect of execution that seems to be getting lost in the discussion and that is running tests locally. As a developer I want to be able to also run the e2e tests locally. For the local development scenario I likely wouldn't run those tests in parallel.